src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -126,9 +126,9 @@ void TryRemoveSlow(IEnumerable c)
             }
         }
 
-        private static ArrayList ToArrayList(IEnumerable items)
+        private static List<object> ToList(IEnumerable items)
         {
-            var list = items is ICollection ic ? new ArrayList(ic.Count) : new ArrayList();
+            var list = items is ICollection ic ? new List<object>(ic.Count) : new List<object>();
 
             foreach (object o in items)
```

Comment:
Can you use AddRange here?

Reply:
Not yet, unfortunately. AddRange takes an `IEnumerable<T>` but the non-generic CollectionTally currently only receives an `IEnumerable`